contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+				</li>
+				<li>
+					Free the server from managing the user interface =&gt; simplified server component =&gt; enhanced scalability
+				</li>
+				<li>
+					Allow the architectural components to evolve independently =&gt; enhanced modifiability 
+				</li>
+			</ul>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			<p>Stateless</p>
+			<p>(Internet Scale)</p>
+		</td>
+		<td>
+			Communications must be stateless: all the messages must be self-sufficient, containing all the necessary information to understand the message without referring to information outside the message
+		</td>
+		<td>
+			<ul>
+				<li>
+					Other architectural components can monitor the interactions with the self-sufficient messages =&gt; enhanced visibility
+				</li>
+				<li>
+					Free the server from managing different states of resources across requests =&gt; enhanced reliability and scalability
+				</li>
+			</ul>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			<p>Cache</p>
+			<p>(Distributed Hypermedia)</p>
+		</td>
+		<td>
+			Data within the response has to be indicated if it is cacheable. If cacheable, the cache will store the data and reuse if for identical requests later.
+		</td>
+		<td>
+			<ul>
+				<li>
+					Eliminate some interactions between server and client =&gt; improved efficiency
+				</li>
+				<li>
+					Reduce average latency of interactions i.e. faster response =&gt; improved user-perceived performance
+				</li>
+			</ul>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			<p>Layered System</p>
+			<p>(Internet Scale)</p>
+		</td>
+		<td>
+			Multiple layers between client and server: each architectural component in the system does not care what happens beyond the layers it has immediate interactions with
+		</td>
+		<td>
+			<ul>
+				<li>
+					reduce complexity of the system =&gt; enhanced simplicity
+				</li>
+				<li>
+					promote independence of the architectural components =&gt; enhanced modifiability
+				</li>
+				<li>
+					allow for load-balancing intermediaries to distribute the workloads =&gt; enhanced scalability
+				</li>
+			</ul>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			<p>Uniform Interface</p>
+			<p>(Extensibility, Internet Scale, Distributed Hypermedia)</p>
+		</td>
+		<td>
+			A uniform interface between the architectural components, defined by 4 interface constraints
+		</td>
+		<td>
+			<ul>
+				<li>
+					simplify the architecture
+				</li>
+				<li>
+					interactions between the architectural components can be monitored as the set of interactions are predefined =&gt; increased visibility
+				</li>
+				<li>
+					decouple service from the implementation (how the service is produced) =&gt; allow the architectural components to evolve independently, which increases modifiability
+				</li>
+			</ul>
+		</td>
+	</tr>
+</table>
+
+The 4 interface constraints defining the Uniform Interface architectural constraint for REST are elaborated in the table below:
+
+<table>
+	<tr>
+		<th> Interface Constraint </th>
```

Comment:
What's the difference between these constraints and REST constraints above? What are "Interface Constraints"?